### PR TITLE
Switch build system to use CMake and vcpkg

### PR DESCRIPTION
**Pull request by olsen232**
New year, new build

CMake works on Windows, macOS, and Linux - improvement on using a mix of make and Microsoft NMAKE.

vcpkg simplifies building dependencies from a variety of sources with a variety of build systems.

Python script fix_vendor_libs.py validates that the transitive closure of dependencies is reasonable and complete, and that rpaths etc are correctly configured for all dynamic libraries - a more robust approach than fixing them one by one in makefiles.

End-to-end tests now test the packaging step - Kart is packaged, then installed, then the end-to-end tests are run against the installed Kart.

Old build that is still on master up to now is relatively unmaintained for a long time - better to use this and iterate as needed.

### CODE REVIEW

**Review by craigds — APPROVED** [review]
obviously too large to review _in detail_ but I tested the amd64 linux build at least and it looks good to me.

has anyone tried the arm64 (either linux or macos) build?

File: kart.spec [hunk 1]
```diff
@@ -74,43 +94,110 @@ if is_linux or is_darwin:
     assert dylib.exclude_list.search('libodbc.so.1')
     assert dylib.exclude_list.search('libodbc.so.2')
 
-if is_darwin:
-    # on macOS every dylib dependency path gets rewritten to @loader_path/...,
-    # which isn't much use wrt unixODBC. And PyInstaller has no useful hooks.
-    # TODO: when we upgrade PyInstaller this probably needs redoing
-    import macholib.util
+# if is_darwin:
```
[the remark this answers is not in the document]
Comment (olsen232) [reply kart.spec:97]: Done

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -142,83 +107,31 @@ solution file and several project files, and by default it chooses the latest
...
-$ cd kart
-$ make
-
-# check it's working
-$ venv/bin/kart --version
-Kart v0.9.1.dev0, Copyright (c) Kart Contributors
-» GDAL v3.0.4
-» PyGit2 v1.1.0; Libgit2 v0.99.0; Git v2.25.1.windows.1
-» APSW v3.30.1-r3; SQLite v3.30.1; SpatiaLite v5.0.0-beta0
+> cmake -B build -S . -DPython3_ROOT=C:\Program Files\Python310 -DVENDOR_ARCHIVE=D:\path\to\downloaded\vendor-windows-X64-3.10.zip -DUSE_VCPKG=OFF
```
[the remark this answers is not in the document]
Comment (olsen232) [reply CONTRIBUTING.md:123]: Done

File: kart.spec [hunk 3]
```diff
@@ -74,43 +94,110 @@ if is_linux or is_darwin:
...
     hiddenimports=[
+        # TODO: improve this somehow
+        *collect_submodules('kart'),
+        *collect_submodules('kart.annotations'),
+        *collect_submodules('kart.lfs_commands'),
+        *collect_submodules('kart.point_cloud'),
+        *collect_submodules('kart.sqlalchemy'),
+        *collect_submodules('kart.sqlalchemy.adapter'),
+        *collect_submodules('kart.tabular'),
+        *collect_submodules('kart.upgrade'),
```
[the remark this answers is not in the document]
Comment (olsen232) [reply kart.spec:200]: Not sure. It has a `# TODO - improve this somehow` on it so I'll leave it for now but would be good to at least document why it's like this

File: kart/checkout.py [hunk 4]
```diff
@@ -185,6 +188,29 @@ def checkout(
         )
 
 
+def _git_fetch_supports_flag(repo, flag):
```
[the remark this answers is not in the document]
Comment (olsen232) [reply kart/checkout.py:191]: Done

**Pull request merged**